src/arg_parsers/identify_format/identify_format.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{arg_parse_err::ArgParseErr, arg_parsers::identify_format::parser::parse};
+use std::ffi::OsStr;
+
+// https://imagemagick.org/script/escape.php
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Var {
+    Colorspace,
+    CurrentImageHeightInPixels,
+    CurrentImageWidthInPixels,
+    ImageDepth,
+    ImageFileFormat,
+    ImageFilename,
+    LayerCanvasPageGeometry,
+    MagickFilename,
+    OriginalImageSize,
+    PageCanvasHeight,
+    PageCanvasWidth,
+    PageCanvasXOffset,
+    PageCanvasYOffset,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Token {
+    Literal(String),
+    Var(Var),
+}
+
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct IdentifyFormat {
+    pub template: Option<Vec<Token>>,
```

Comment:
Could you move the `Option` out of the `IdentifyFormat` struct, so that the call sites refer to not just `IdentifyFormat` but `Option<IdentifyFormat>`? This is more boilerplate, but I really prefer for optionals to be explicit.